core/src/main/java/feast/core/model/Entity.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2019 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.core.model;
+
+import feast.core.FeatureSetProto.EntitySpec;
+import feast.types.ValueProto.ValueType;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@javax.persistence.Entity
+public class Entity extends Field {
+
+  @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+  @JoinColumns({
+    @JoinColumn(name = "project", referencedColumnName = "project"),
+    @JoinColumn(name = "feature_set", referencedColumnName = "feature_set"),
+    @JoinColumn(name = "version", referencedColumnName = "version"),
+    @JoinColumn(name = "name", referencedColumnName = "name")
+  })
+  private List<Statistics> statistics;
+
+  public Entity() {}
+
+  public Entity(String name, ValueType.Enum type) {
+    this.setId(new FieldId());
```

Comment:
Would it make more sense, safety wise, to make all of the fields in `FieldId` required (non-nullable), and then to only set this `FieldId` once we have all the information available to build it? I am thinking that can happen during `fs1.add(field1)` after which the field can then be persisted.

Reply:
And duplicate the `name` in the field itself?

Reply:
Oops. The test failed because I forgot to populate the field. I think it's safe enough to just enforce the constraint on the DB side.

Reply:
> And duplicate the `name` in the field itself?

Not sure what you are asking